File: BOJ/구현/5073.py
# 5073 삼각형과 세 변
# https://www.acmicpc.net/problem/5073
import sys
input = sys.stdin.readline

while True:
    a, b, c = map(int, input().split())

    if a == 0 and b == 0 and c == 0:
        break

    if b + c <= a or a + c <= b or a + b <= c:
        print("Invalid")

    else:
        if a == b and b == c:
            print("Equilateral")
        elif (a == b and b != c) or (b == c and c != a) or (a == c and b != c):
            print("Isosceles")
        else:
            print("Scalene")

File: BOJ/구현/5073/5073.cpp
//  5073 삼각형과 세 변
//  https://www.acmicpc.net/problem/5073
#include <iostream>
#include <cmath>

using namespace std;

int main(void) {
  ios_base::sync_with_stdio(false); 
  cin.tie(NULL); 
  cout.tie(NULL);

  while (1) {
    int a, b, c;
    cin >> a >> b >> c;
    
    if (a == 0 && b == 0 && c == 0) 
      break;
    else {
      if (a+b <= c || b+c<=a || a+c <= b) cout << "Invalid" << "\n";
      else {
        if (a == b && b == c) {
          cout << "Equilateral" << "\n";
        } else if ((a == b && b != c) || (b == c && a != c) || (a == c && c != b)) {
          cout << "Isosceles" << "\n";
        } else {
          cout << "Scalene" << "\n";
        }
      }
    }
  }

  return 0;
}